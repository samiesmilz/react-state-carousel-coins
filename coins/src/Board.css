.Board {
  padding: 25px;
  text-align: center;
  border: 1px solid cadetblue;
  padding-bottom: 50px;
  border-radius: 15px;
  margin: 10px;
}
