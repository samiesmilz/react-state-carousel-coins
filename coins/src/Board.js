import React, { useState } from "react";
import "./Board.css";
import Coin from "./Coin.js";
import images from "./Images.js";

const Board = () => {
  const [side, setSide] = useState(images[0]);
  const [counter, setCounter] = useState(0);
  const [heads, setHeads] = useState(0);
  const [tails, setTails] = useState(0);
  const [isVisible, setIsVisble] = useState(false);

  const flipCoin = () => {
    const flipResult = Math.floor(Math.random() * images.length);
    setSide(images[flipResult]);
    setCounter(counter + 1);
    flipResult === 0 ? setHeads(heads + 1) : setTails(tails + 1);
    setIsVisble(true);
  };

  return (
    <div className="Board">
      <h1>Let's flip a coin!</h1>
      {isVisible && <Coin side={side} />}
      <h2>Stats:</h2>
      <p>
        <strong>Heads:</strong> {heads}
      </p>
      <p>
        <strong>Tails:</strong> {tails}
      </p>
      <p>
        Out of {counter} flips: There are {heads} heads and {tails} tails.
      </p>
      <button onClick={flipCoin}>Flip Coin</button>
    </div>
  );
};

export default Board;
