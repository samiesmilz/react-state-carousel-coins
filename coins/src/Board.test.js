import React from "react";
import { render } from "@testing-library/react";
import Board from "./Board.js";

it("renders without crashing...", () => {
  render(<Board />);
});

it("matches snapshot...", () => {
  const { asFragment } = render(<Board />);
  expect(asFragment()).toMatchSnapshot();
});
