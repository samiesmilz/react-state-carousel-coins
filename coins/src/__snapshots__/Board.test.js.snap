// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot... 1`] = `
<DocumentFragment>
  <div
    class="Board"
  >
    <h1>
      Let's flip a coin!
    </h1>
    <h2>
      Stats:
    </h2>
    <p>
      <strong>
        Heads:
      </strong>
       0
    </p>
    <p>
      <strong>
        Tails:
      </strong>
       0
    </p>
    <p>
      Out of 0 flips: There are 0 heads and 0 tails.
    </p>
    <button>
      Flip Coin
    </button>
  </div>
</DocumentFragment>
`;
